### ISSUE TITLE
Operator-Linebreak Default Option Behaves Differently When Option Is Actually Specified

### ISSUE THREAD

**Issue by epmatsw** [eslint/eslint#7459]
**Tell us about your environment**
- **ESLint Version:** 3.8.1
- **Node Version:** 7.0.0
- **npm Version:** 3.10.9

**What parser (default, Babel-ESLint, etc.) are you using?** default

**Please show your full configuration:**

A

```
{
    "rules": {
        "operator-linebreak": [2, "after"]
    }
}
```

B

```
{
    "rules": {
        "operator-linebreak": [2]
    }
}
```

**What did you do? Please include the actual source code causing the issue.**

``` js
var blah = hey ?
    a : b;
```

**What did you expect to happen?**
`"after" (default) requires linebreaks to be placed after the operator (except for the ternary operator characters ? and :)`

Based on the docs, I would expect this code to pass with both configs.
**What actually happened? Please include the actual, raw output from ESLint.**
If you use config B, the line is flagged as an error. Config A passes as expected.

I think this is really a doc issue. It should be more clear that the `after` option and the `default` behavior are not interchangeable.

**Comment by not-an-aardvark** [comment]
Thanks for the report. I was going to say that this is a bug, but based on the [source code](https://github.com/eslint/eslint/blob/ee3bcea71fe8f826e7c0494f5388ad80542fbc35/lib/rules/operator-linebreak.js#L46-L57) it appears to be intentional. I'm not sure why, though.

If it is intentional, then I agree that the documentation should point it out.

**Comment by epmatsw** [comment]
Yeah, it definitely feels buggy. And even the docs in that commit seem unclear on what the behavior should be.

**Comment by not-an-aardvark** [comment]
I figured it out: it's not that there's a bug in the rule, it's that the default configuration has overrides for `?` and `:`. In other words, the default configuration is not just `after`, it's `after, {overrides: {'?': before, ':': before}}`.

I made a PR to update the docs at https://github.com/eslint/eslint/pull/7726.
